content-scripts/cs.js

```diff
@@ -89,43 +89,48 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
   console.log("[Message from background]", request);
   if (request === "getInitialUrl") {
     sendResponse(initialUrl);
-  } else if (request.themesUpdated) {
-    injectUserstylesAndThemes({ themes: request.themesUpdated, isUpdate: true });
   }
 });
 
 // Store all themes that were enabled this session
 const sessionEnabledThemes = new Set();
 
-function injectUserstylesAndThemes({ addonsWithUserstyles = [], themes, isUpdate }) {
-  document.querySelectorAll(".scratch-addons-theme").forEach((style) => {
-    if (!style.textContent.startsWith("/* sa-autoupdate-theme-ignore */")) style.remove();
-  });
-  const userstyles = addonsWithUserstyles.map((addon) => addon.styles);
-  for (const addon of userstyles || []) {
-    for (const userstyle of addon) {
+function addStyle(addon) {
+  for (let userstyle of addon.styles) {
+    if (addon.injectAsStyleElt) {
+      sessionEnabledThemes.add(addon.addonId);
+
+      // Replace %addon-self-dir% for relative URLs
+      userstyle = userstyle.replace(/\%addon-self-dir\%/g, chrome.runtime.getURL(`addons/${addon.addonId}`));
+      //userstyle += `\n/*# sourceURL=${styleUrl} */`;
```

Comment:
Why was this commented? Is the `styleUrl` not available? It should, so that we can provide a source map

Reply:
commented for now, until i send styleUrl back, it was removed bc we merged userstyles and themes.